app/src/main/java/com/mobilecoronatracker/data/repository/impl/CountriesDataRepoImpl.kt

```diff
@@ -0,0 +1,77 @@
+package com.mobilecoronatracker.data.repository.impl
+
+import com.mobilecoronatracker.data.persistence.AppDatabase
+import com.mobilecoronatracker.data.persistence.dao.CountryDao
+import com.mobilecoronatracker.data.persistence.dao.CountryDataDao
+import com.mobilecoronatracker.data.persistence.entity.Country
+import com.mobilecoronatracker.data.persistence.entity.CountryData
+import com.mobilecoronatracker.data.repository.CountriesDataRepo
+import com.mobilecoronatracker.data.repository.CovidDataRepo
+import com.mobilecoronatracker.model.CountryReportModelable
+import com.mobilecoronatracker.model.impl.CountryReportModel
+import com.mobilecoronatracker.utils.getTodayTimestamp
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class CountriesDataRepoImpl(
+    private val countryDataDao: CountryDataDao,
+    private val countryDao: CountryDao,
+    private val covidDataRepo: CovidDataRepo,
+    private val appDatabase: AppDatabase
+) : CountriesDataRepo {
+    override fun getAllCountriesTodayData(): Flow<List<CountryReportModelable>> {
+        val todayTimestamp = getTodayTimestamp()
+        return countryDataDao.getAllCountryByTimestampFlow(todayTimestamp).map { list ->
+            list.map {
+                CountryReportModel(it)
+            }
+        }
+    }
+
+    override suspend fun hasNoTodayCountryData(): Boolean {
+        val todayTimestamp = getTodayTimestamp()
+        return countryDataDao.getAllCountryByTimestamp(todayTimestamp).map {
```

Comment:
To get all, then map and check if result is empty seems to be a bit of an overkill to me. Maybe you could extend dao with getCount() method linked to a query like this: SELECT count(*) FROM country_data? I think it is better to delegate this to db engine.

Reply:
Right, I was a little bit too lazy here ;) fixed